animal-options.lua

```diff
@@ -0,0 +1,175 @@
+function creature_screen:set_adoption_flag(option)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if not unit then return end
+
+    if option == YES then
+        unit.flags3.available_for_adoption = true
+    else
+        unit.flags3.available_for_adoption = false
+    end
+end
+
+-- Check current flag status of animal to dynamically set on/off
+function creature_screen:get_butcher(unit)
+    if not unit then return end
+
+    if unit.flags2.slaughter == true then
+        return YES
+    else
+        return NO
+    end
+end
+
+function creature_screen:get_geld(unit)
+    if not unit then return end
+
+    if unit.flags3.marked_for_gelding == true then
+        return YES
+    else
+        return NO
+    end
+end
+
+function creature_screen:get_adopt(unit)
+    if not unit then return end
+
+    if unit.flags3.available_for_adoption == true then
+        return YES
+    else
+        return NO
+    end
+end
+
+-- Use render to set On/Off dynamically for each unit
+function creature_screen:render(dc)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    self.subviews.butcher_animal:setOption(self:get_butcher(unit))
+    self.subviews.geld_animal:setOption(self:get_geld(unit))
+    self.subviews.adopt_animal:setOption(self:get_adopt(unit))
+
+    creature_screen.super.render(self, dc)
+end
+
+function creature_screen:init()
+self:addviews{
+    widgets.Panel{
+        frame_background=gui.CLEAR_PEN,
+        frame_style=gui.FRAME_MEDIUM,
+        visible=check_valid_unit,
+        subviews={
+            widgets.CycleHotkeyLabel{
+                frame={t=0,l=0},
+                label='Butcher',
+                key='CUSTOM_CTRL_B',
+                options={
+                    {label=NO, value=NO, pen=COLOR_WHITE},
+                    {label=YES, value=YES, pen=COLOR_GREEN},
+                },
+                view_id='butcher_animal',
+                on_change=function(val) self:set_slaughter_flag(val) end,
+            },
+            widgets.ToggleHotkeyLabel{
+                frame={t=1,l=0},
+                label='Geld',
+                key='CUSTOM_CTRL_G',
+                options={
+                    {label=NO, value=NO, pen=COLOR_WHITE},
+                    {label=YES, value=YES, pen=COLOR_GREEN},
+                },
+                view_id='geld_animal',
+                enabled=is_geldable,
+                on_change = function(val) self:set_geld_flag(val) end
+            },
+            widgets.ToggleHotkeyLabel{
+                frame={t=2,l=0},
+                label='Adopt',
+                key='CUSTOM_CTRL_A',
+                options={
+                    {label=NO, value=NO, pen=COLOR_WHITE},
+                    {label=YES, value=YES, pen=COLOR_GREEN},
+                },
+                view_id='adopt_animal',
+                on_change = function(val) self:set_adoption_flag(val) end
+            },
+        },
+    },
+}
+end
+
+OVERLAY_WIDGETS = {
+    creaturescreen=creature_screen,
```

Comment:
how about `animal_actions` as the widget name?

Reply:
Made change